### PR TITLE
Adds Greenscale Coloration to Night Vision Goggles

### PR DESCRIPTION
**Pull request by Shadowmech88**
Night vision goggles now make the wearer's vision greenscale while active.
Night vision goggles have been made into a child of `scanner`, and can now be toggled on or off.

Credit to @DrCelt for the specific color matrix the goggles use, as color matrices are still mostly a mystery to me.

:cl:
 * rscadd: Added a green tint to night vision goggles.

### CODE REVIEW

**Comment by jakmak3** [comment]
The tint is a little much. Maybe reduce its severity by a small amount?

**Comment by Shadowmech88** [comment]
Since it's greenscale, all reducing the tint would do is desaturate the green.

**Comment by gbasood** [comment]
Can you not just increase the values for other colors in the color matrix?

**Comment by Shadowmech88** [comment]
Well yeah, but then it wouldn't be greenscale anymore, it'd just be normal vision with a layer of green over everything.
Actual NVGs can only display varying brightnesses of green, other colors are actually not present in its display.

File: vgstation13.dme [hunk 1]
```diff
@@ -2053,7 +2053,7 @@
 #include "interface\web\interface.dms"
 #include "maprendering\maprendering.dm"
 #include "maps\_map.dm"
-#include "maps\tgstation.dm"
+#include "maps\test_tiny.dm"
```
Comment (ghost) [inline vgstation13.dme]: Whoops

**Comment by clusterfack** [comment]
You could probably make it so it lights up dark areas with a different shade of green than light areas, by modifying the client color matrix as you did, along with a screen object on the lighting plane.

**Comment by Zth--** [comment]
Is that character Trump?

**Comment by SysadminJeroen** [comment]
@Zth-- Yes, look closer.

**Comment by SonixApache** [comment]
Could you check what this does for combat borgs now that #12485 got merged?

**Comment by Shadowmech88** [comment]
I'd bet money that it does nothing for borgs, since the color matrix application is in the code for equipping the goggles themselves.

It'd be easy enough to add the color matrix to that borg module, though.

**Comment by Intigracy** [comment]
If you do I'll merge this.

**Comment by Shadowmech88** [comment]
Done.

**Comment by ComicIronic** [comment]
You should really have made the colour matrix global and then referenced it in two places, rather than copy-pasting.

**Pull request merged**